### PR TITLE
Feature/dimensions component [DHIS2-3280]

### PR DESCRIPTION
**Pull request by joakimia**
This feature adds the list of available dimensions to the left panel within the data visualizer.
It enables users to configure what type of output which should be taken into account when visualising data.

When clicking on a dimension, a popup dialog will be displayed and enable configuration of the given dimension.
There are currently no dimension components available (the Chart Options popup-dialog are being used as a placeholder).

The first three dimensions "Data, Period, Organisation Units" are not present in the API call - these dimensions are added with the rest received from the redux store.

As of now, no actions are dispatched to the store, but this can change given the response from reviewers on this PR.

There's also a <Star/> placeholder icon used as of now, correct icons will be retrieved from material-ui-icons.

More tests will also be added during the review.

### CODE REVIEW

File: src/Dimensions/DimensionsManager.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import i18n from '@dhis2/d2-i18n';
+
+import VisualizationOptions from '../VisualizationOptions/VisualizationOptions';
+
+// Placeholder for the dimension popup dialogs - using the Options dialog until the components are created
+let dimensionsArr = [<VisualizationOptions />, 'test'];
+
+export const DimensionsManager = ({ index, dialogIsOpen, toggleDialog }) => {
+    return (
+        <Fragment>
```
Comment (Stevenah) [inline src/Dimensions/DimensionsManager.js]: Is Fragment needed here?

File: src/Dimensions/Dimensions.js [hunk 2]
```diff
@@ -1,16 +1,90 @@
...
-};
+export class Dimensions extends Component {
+    state = {
+        searchFieldValue: '',
+        dialogIsOpen: false,
+        dimensions: ['Data', 'Periods', 'Organisation Units'],
+    };
+
+    onClick = index => {
+        //this.props.tSetDimension('dimension', index); // (dispatch, getState)
```
Comment (Stevenah) [inline src/Dimensions/Dimensions.js]: Remove comment

File: src/Dimensions/Alternatives.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import List, { ListItem, ListItemIcon, ListItemText } from 'material-ui/List';
+import { Data } from './icons/data';
+import { Period } from './icons/period';
+import { OrgUnit } from './icons/orgunit';
+import { GenericDimension } from './icons/genericdimension';
+
+// import { Data, Period, OrgUnit, GenericDimension } from './icons';  resulting in module not found (?)
```
Comment (Stevenah) [inline src/Dimensions/Alternatives.js]: Remove comment

File: src/Dimensions/Alternatives.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    textStyle: {
+        fontSize: 16,
+        paddingLeft: 10,
+        paddingRight: 0,
+    },
+};
+
+const dimensionIcon = [<Data />, <Period />, <OrgUnit />];
+
+export const Alternatives = ({ dimensions, searchFieldValue, onClick }) => {
```
Comment (Stevenah) [inline src/Dimensions/Alternatives.js]: Should clean this up by making it a class.

File: src/Dimensions/Dimensions.js [hunk 5]
```diff
@@ -1,16 +1,90 @@
...
+    toggleDialog = () => {
+        this.setState({ dialogIsOpen: !this.state.dialogIsOpen });
+    };
+
+    componentWillReceiveProps = nextProps => {
+        this.setState({
+            dimensions: [
+                ...this.state.dimensions,
+                ...Object.entries(nextProps.dimensions).map(
+                    entry => entry[1].displayName
```
Comment (neeilya) [inline src/Dimensions/Dimensions.js]: Just a small thing, consider using
```
Object.entries(nextProps.dimensions).forEach(([key, entry]) => entry.displayName);
```
to get rid of the `[1]` index :)

**Comment by neeilya** [comment]
Approved, but encourage some more developers to review :)

File: src/api/dimensions.js [hunk 6]
```diff
@@ -4,5 +4,9 @@ import { onError } from './index';
 // Get dimensions on startup
 export const apiFetchDimensions = () =>
     getInstance()
-        .then(d2 => d2.Api.getApi().get('dimensions.json', { paging: false }))
+        .then(d2 =>
+            d2.Api.getApi().get('dimensions.json', {
```
Comment (jenniferarnesen) [inline src/api/dimensions.js]: Is it possible to fetch dimensions with the d2 Models api instead?  (e.g, something like d2.models.list('dimensions').

Comment (edoardo) [reply src/api/dimensions.js]: The syntax is something like: `const dimensions = d2.models.dimensions.list();` which returns a ModelCollection object.

File: src/Dimensions/DimensionOptions.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    };
+
+    render = () => {
+        let Options = <OptionsButton action={this.showMenu} />;
+
+        if (this.state.showMenu) {
+            Options = this.renderDropDown();
+        }
+
+        return <div>{Options}</div>;
```
Comment (edoardo) [inline src/Dimensions/DimensionOptions.js]: Can this div be removed?

**Pull request merged**